Remora.Results/Result.cs

```diff
@@ -32,16 +32,18 @@ namespace Remora.Results;
 [PublicAPI]
 public readonly struct Result : IResult
 {
+    /// <summary>
+    /// Gets a successful result.
+    /// </summary>
+    public static Result Success => FromSuccess();
```

Comment:
You should cache this result as well, same as with the other one.

Reply:
Won't accessing a `Result Success { get; } = ...` property return a copy of the struct anyway?

Reply:
A copy, yes, but a copy is not the same as a new instance. The compiler can make other optimization decisions when doing copies, while creating an entirely new object has additional requirements.

Reply:
Fixed.